app/src/main/java/cz/movapp/app/ui/exercise/setup/ExerciseSetupFragment.kt

```diff
@@ -0,0 +1,89 @@
+package cz.movapp.app.ui.exercise.setup
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import cz.movapp.app.BuildConfig
+import cz.movapp.app.R
+import cz.movapp.app.databinding.FragmentExerciseSetupBinding
+import cz.movapp.app.ui.dictionary.DictionaryMetaCategoryData
+import cz.movapp.app.ui.dictionary.DictionaryViewModel
+import cz.movapp.app.ui.exercise.ExerciseViewModel
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+class ExerciseSetupFragment : Fragment() {
+
+    private lateinit var binding: FragmentExerciseSetupBinding
+
+    private val dictionarySharedViewModel: DictionaryViewModel by activityViewModels()
+    private val exerciseViewModel: ExerciseViewModel by activityViewModels()
+
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentExerciseSetupBinding.inflate(inflater, container, false)
+
+        binding.apply {
+            start.setOnClickListener {
+                lifecycleScope.launch {
+                    val selectedCategories = (binding.recyclerViewExercise.adapter as ExerciseSetupAdapter).selectedCategories()
+                    val numExercises = binding.groupLength.checkedRadioButtonId.numExercises()
+                    val error = when {
+                        selectedCategories.isEmpty() -> getString(R.string.exercise_setup_error_select_categories)
+                        numExercises == 0 -> getString(R.string.exercise_setup_error_select_length)
+                        else -> null
+                    }
+                    if (error == null) {
+                        binding.error.visibility = View.GONE
+                        exerciseViewModel.prepareExercise(selectedCategories, numExercises)
+                        exerciseViewModel.startNewExercise()
+                        findNavController().navigate(R.id.to_play)
+                    }
+                    else {
+                        binding.error.visibility = View.VISIBLE
+                        binding.error.text = error
+                    }
+                }
+            }
+        }
+
+        dictionarySharedViewModel.metaCategories.observe(viewLifecycleOwner) {
+            if (BuildConfig.DEBUG) {
```

Comment:
Myslím, že to není třeba obalovat tím ifem. V App.kt je už podmínka debug modu vyřešená.